File: dominio/modelos orijinales/NavigatorMenu.java
package servicebackmenu2;


abstract class NavigatorMenu extends MessagesMenu{
       
    private String nameActualMenu;
    private Menu actualMenu;
    private ListMenus listMenus;
    private NavigatorOption navigatorOption;
    
    public NavigatorMenu(String title) {
        super(title);
        this.nameActualMenu = this.title;
        this.actualMenu = this;
        this.listMenus = ListMenus.getInstance();
        this.listMenus.addMenu(this.nameActualMenu, this.actualMenu);
        this.navigatorOption = new NavigatorOption();
    }

    protected void showTitles() {
        this.add(this.navigatorOption);
        super.showTitles();
    }

}

File: conect4/src/main/resources/static/js/memus/models/generics/NavigatorMenu.js

import Menu from "./Menu.js";
import ListMenus from "./ListMenus.js";
import NavigatorOption from "./NavigatorOption.js";

class NavigatorMenu extends Menu {
    
    nameActualMenu;
    actualMenu;
    listMenus;
    navigatorOption;
    
    constructor() {
        super();
        this.nameActualMenu = this.constructor.name;
        this.actualMenu = this.getMenu();
        this.listMenus = ListMenus.getInstance();
        this.listMenus.addMenu(this.nameActualMenu, this.actualMenu);
        this.navigatorOption = new NavigatorOption();
    }

    addOptions() {
        this.add(this.navigatorOption);
    }
    
}

export default NavigatorMenu;